pyxem/utils/ransac_ellipse_tools.py

```diff
@@ -574,3 +576,118 @@ def _get_ellipse_markers(
         marker_list.extend(marker_in_list)
         marker_list.extend(marker_out_list)
     return marker_list
+
+
+def get_max_positions(signal,
+                      mask=None,
+                      num_points=5000,
+                      ):
+    """ Gets the top num_points pixels in the dataset.
+
+    Parameters
+    --------------
+    signal: BaseSignal
+        The signal which we want to find the max positions for.
+    mask: np.array
+        A mask to be applied to the data for values to ignore
+    num_points: int
+        The number of points to be
+    """
+    if isinstance(signal, BaseSignal):
+        data = signal.data
+    else:
+        data = signal
+    i_shape = np.shape(data)
+    flattened_array = data.flatten()
+    if mask is not None:
+        flattened_mask = mask.flatten()
+        flattened_array[flattened_mask]=0
+    # take top 5000 points make sure exclude zero beam
+    indexes = np.argsort(flattened_array)
+    cords = np.array([np.floor_divide(indexes[-num_points:], i_shape[1]),
+             np.remainder(indexes[-num_points:], i_shape[1])]) # [x axis (row),y axis (col)]
+    return cords.T
+
+
+def determine_ellipse(signal,
+                      mask=None,
+                      num_points=1000,
+                      use_ransac=True,
+                      guess_starting_params=True,
+                      **kwargs,
+                      ):
+    """
+    This method starts by taking some number of points which are the most intense
+    in the signal.  It then takes those points and guesses some starting parameters
+    for the `get_ellipse_model_ransac_single_frame` function. From there it will try
+    to determine the ellipse parameters.
+
+    Parameters
+    -----------
+    signal: Signal2D
+        The signal of interest
+    mask: Array-like
+        The mask to be applied to the data.  All of the masked values are ignored
```

Comment:
Write if `True` or `False` is considered the mask. Would be nice with an example showing how to mask the centre beam, further down in the docstring.